### PR TITLE
CI Provider info selector logic to internal-api

### PR DESCRIPTION
**Pull request by drodriguezhdez**
This PR moves the `CIProviderInfo` logic to the `internal-api` module.

### CODE REVIEW

**Comment by richardstartin** [comment]
I'm not sure we can move something as volatile as this in to public API. Whilst it's unlikely that any tracer user would depend on these, you won't be able to remove them after you merge this.

**Comment by drodriguezhdez** [comment]
These tags keys and logic to obtain the CI provider information are needed to send the "benchmark" tests that we discussed. The CIProvider information logic needs to be accessible for the benchmark tests project. Regarding the tag keys, I think I could duplicate these ones in the datadog-jmh project until everything is more settled.

File: dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/instrumentation/decorator/TestDecoratorTest.groovy [hunk 1]
```diff
@@ -33,6 +33,7 @@ class TestDecoratorTest extends BaseDecoratorTest {
     1 * span.setTag(Tags.BUILD_SOURCE_ROOT, "ci-workspace-path")
     1 * span.setTag(Tags.GIT_REPOSITORY_URL, "git-repository-url")
     1 * span.setTag(Tags.GIT_COMMIT_SHA, "git-commit")
+    1 * span.setTag(Tags._GIT_COMMIT_SHA, "git-commit")
```
Comment (richardstartin) [inline dd-java-agent/agent-bootstrap/src/test/groovy/datadog/trace/bootstrap/instrumentation/decorator/TestDecoratorTest.groovy:36]: what's this? How is it different to the line above?

File: internal-api/src/main/java/datadog/trace/bootstrap/instrumentation/api/Tags.java [hunk 2]
```diff
@@ -44,7 +44,8 @@
   public static final String BUILD_SOURCE_ROOT = "build.source_root";
 
   public static final String GIT_REPOSITORY_URL = "git.repository_url";
-  public static final String GIT_COMMIT_SHA = "git.commit_sha";
+  public static final String GIT_COMMIT_SHA = "git.commit.sha";
+  @Deprecated public static final String _GIT_COMMIT_SHA = "git.commit_sha";
```
Comment (richardstartin) [inline internal-api/src/main/java/datadog/trace/bootstrap/instrumentation/api/Tags.java:48]: ditto

**Review by richardstartin — APPROVED** [review]

**Review by mcculls — APPROVED** [review]

File: internal-api/src/main/java/datadog/trace/bootstrap/instrumentation/api/Tags.java (hunk 2 above)
Comment (drodriguezhdez) [reply internal-api/src/main/java/datadog/trace/bootstrap/instrumentation/api/Tags.java:48]: We updated the `git.commit` tags to support author, committer and message, but we need to maintain the previous one until backend was updated.

**Pull request merged**